pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -219,20 +230,89 @@ func (a ClusterOperatorAnonymizer) Marshal(_ context.Context) ([]byte, error) {
 	return runtime.Encode(serializer, a.ClusterOperator)
 }
 
-// type ClusterOperatorListAnonymizer struct{ *configv1.ClusterOperatorList }
+func isFailingOperator(operator *configv1.ClusterOperator) bool {
+	for _, condition := range operator.Status.Conditions {
+		switch {
+		case condition.Type == configv1.OperatorDegraded && condition.Status == configv1.ConditionTrue,
+			condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionFalse:
+			return true
+		}
+	}
+	return false
+}
+
+func namespacesForOperator(operator *configv1.ClusterOperator) []string {
+	var ns []string
+	for _, ref := range operator.Status.RelatedObjects {
+		if ref.Resource == "namespaces" {
+			ns = append(ns, ref.Name)
+		}
+	}
+	return ns
+}
 
-// func (a ClusterOperatorListAnonymizer) Marshal(_ context.Context) ([]byte, error) {
-// 	return runtime.Encode(serializer, a.ClusterOperatorList)
-// }
+type NodeAnonymizer struct{ *corev1.Node }
 
-// func clusterOperatorListResourceVersion(items *configv1.ClusterOperatorList) string {
-// 	rvs := make([]string, 0, len(items.Items))
-// 	for _, item := range items.Items {
-// 		rvs = append(rvs, item.ResourceVersion)
-// 	}
-// 	sort.Strings(rvs)
-// 	return strings.Join(rvs, ",")
-// }
+func (a NodeAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	return runtime.Encode(kubeSerializer, anonymizeNode(a.Node))
+}
+
+func anonymizeNode(node *corev1.Node) *corev1.Node {
+	for k := range node.Annotations {
+		node.Annotations[k] = ""
+	}
+	for k, v := range node.Labels {
+		node.Labels[k] = strings.Repeat("x", len(v))
+	}
+	node.Status.Addresses = nil
+	node.Status.DaemonEndpoints = corev1.NodeDaemonEndpoints{}
+	node.Status.NodeInfo.BootID = strings.Repeat("x", len(node.Status.NodeInfo.BootID))
+	node.Status.NodeInfo.SystemUUID = strings.Repeat("x", len(node.Status.NodeInfo.SystemUUID))
+	node.Status.NodeInfo.MachineID = strings.Repeat("x", len(node.Status.NodeInfo.MachineID))
+	node.Status.Images = nil
+	return node
+}
+
+func isFailingNode(node *corev1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionFalse {
+			return true
+		}
+	}
+	return false
+}
+
+type PodAnonymizer struct{ *corev1.Pod }
+
+func (a PodAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	return runtime.Encode(kubeSerializer, anonymizePod(a.Pod))
+}
+
+func anonymizePod(pod *corev1.Pod) *corev1.Pod {
+	// pods gathered from openshift namespaces and cluster operators are expected to be under our control and contain
+	// no sensitive information
```

Comment:
So far, we have considered the cluster name/url a sensitive information, which is contained in `nodeName` field

Reply:
Another place where the urls are visible are in node selectors and potentially in args for the pod, such as:

```
"args":["start","--apiserver-url=https://api-int.my-cluster.example.com:6443"]
```

For this particular case of the hostnames, given we know the value for the hostame, doing some stupid sed on the raw serialized string might be the safest way to avoid this

Reply:
Node name and cluster URL won't be sensitive going forward.  Other data in the node object will be.  We have to have node name from alerts and from telemetry.